### PR TITLE
Add tests for Intervention IRS plan

### PR DESCRIPTION
**Pull request by kelvin-muchiri**
Add tests to ensure draft and active plans are rendered correctly. The snapshots generated always match the spinner and therefore were not added. Jest does not wait for the async processes to complete despite flushing promises

Addresses #664

### CODE REVIEW

File: server/src/index.ts [hunk 1]
```diff
@@ -132,7 +132,7 @@ const oauthCallback = (req: express.Request, res: express.Response, next: expres
           apiResponse.oAuth2Data = user.data;
 
           const sessionState = getOpenSRPUserInfo(apiResponse);
-          if (sessionState) {
+          if (sessionState && sessionState.extraData && req.session) {
```
Comment (moshthepitt) [inline server/src/index.ts:135]: What caused the need for this?

File: src/store/ducks/generic/jurisdictions.ts [hunk 2]
```diff
@@ -184,7 +184,7 @@ export type ImmutableGenericJurisdictionState = FullGenericJurisdictionState &
   SeamlessImmutable.ImmutableObject<FullGenericJurisdictionState>;
 
 /** initial GenericJurisdiction state */
-const initialState: ImmutableGenericJurisdictionState = SeamlessImmutable({});
+const initialState: any = SeamlessImmutable({});
```
Comment (moshthepitt) [inline src/store/ducks/generic/jurisdictions.ts:187]: Why any?

File: src/store/ducks/generic/plans.ts [hunk 3]
```diff
@@ -95,7 +95,7 @@ interface IRSPlanState {
 export type ImmutableIRSPlanState = SeamlessImmutable.ImmutableObject<IRSPlanState> & IRSPlanState;
 
 /** initial IRSPlan state */
-const initialState: ImmutableIRSPlanState = SeamlessImmutable({
+const initialState: any = SeamlessImmutable({
```
Comment (moshthepitt) [inline src/store/ducks/generic/plans.ts:98]: Can we avoid any?

File: src/store/ducks/generic/structures.ts [hunk 4]
```diff
@@ -148,7 +148,7 @@ export type ImmutableGenericStructureState = FullGenericStructureState &
   SeamlessImmutable.ImmutableObject<FullGenericStructureState>;
 
 /** initial GenericStructure state */
-const initialState: ImmutableGenericStructureState = SeamlessImmutable({});
+const initialState: any = SeamlessImmutable({});
```
Comment (moshthepitt) [inline src/store/ducks/generic/structures.ts:151]: Can we avoid any?

File: src/store/ducks/goals.ts [hunk 5]
```diff
@@ -71,7 +71,7 @@ interface GoalState {
 export type ImmutableGoalState = GoalState & SeamlessImmutable.ImmutableObject<GoalState>;
 
 /** initial Goal state */
-const initialState: ImmutableGoalState = SeamlessImmutable({
+const initialState: any = SeamlessImmutable({
```
Comment (moshthepitt) [inline src/store/ducks/goals.ts:74]: Can we avoid any?

File: server/src/index.ts (hunk 1 above)
Comment (kelvin-muchiri) [reply server/src/index.ts:135]: tslint

File: src/store/ducks/generic/jurisdictions.ts (hunk 2 above)
Comment (kelvin-muchiri) [reply src/store/ducks/generic/jurisdictions.ts:187]: The app faile dot start. I cherry picked https://github.com/onaio/reveal-frontend/commit/2b75e40a03b4b31f5825a6ac5b13f9ad6bbaf891

Comment (p-netm) [reply src/store/ducks/generic/jurisdictions.ts:187]: @kelvin-muchiri This has been resolved in master, well at least in a tad better way than using any.

File: src/containers/pages/InterventionPlan/IRS/plan/tests/index.test.tsx [hunk 6]
```diff
@@ -117,4 +136,90 @@ describe('containers/pages/IRS/plan', () => {
...
+        <Router history={history}>
+          <ConnectedIrsPlan {...props} />
+        </Router>
+      </Provider>
+    );
+    await flushPromises();
+    /**@todo Investigate why match snapshot is not awaiting for async
+     * The snapshot returned is that of Loading component despite execution
+     * proceeding past this
+     */
```
Comment (p-netm) [inline src/containers/pages/InterventionPlan/IRS/plan/tests/index.test.tsx:176]: Everything's looking great, also IMO it seems we should have the investigation of why this happening  and its resolution in this pr.
 cc @cKellyDesign @moshthepitt thoughts?

Comment (moshthepitt) [reply src/containers/pages/InterventionPlan/IRS/plan/tests/index.test.tsx:176]: @kelvin-muchiri can you create an issue for that investigation?

File: src/store/ducks/generic/jurisdictions.ts (hunk 2 above)
Comment (moshthepitt) [reply src/store/ducks/generic/jurisdictions.ts:187]: @p-netm @kelvin-muchiri can we then clean up this PR to match what is in master?

**Pull request merged**